controllers/vrg_kubeobjects.go

```diff
@@ -348,22 +355,96 @@ func (v *VRGInstance) vrgObjectProtect(result *ctrl.Result, s3StoreAccessors []s
 	}
 }
 
+func RecipeInfoExistsOnVRG(vrgInstance ramen.VolumeReplicationGroup) bool {
+	return vrgInstance.Spec.KubeObjectProtection != nil &&
+		vrgInstance.Spec.KubeObjectProtection.Recipe != nil &&
+		vrgInstance.Spec.KubeObjectProtection.Recipe.Name != nil
+}
+
+func VolumeGroupNameExistsInWorkflow(vrgInstance ramen.VolumeReplicationGroup) bool {
+	return vrgInstance.Spec.KubeObjectProtection.Recipe.Workflow != nil &&
+		vrgInstance.Spec.KubeObjectProtection.Recipe.Workflow.VolumeGroupName != nil
+}
+
+func GetLabelSelectorFromRecipeVolumeGroupWithName(name string, recipe *Recipe.Recipe) (metav1.LabelSelector, error) {
+	labelSelector := &metav1.LabelSelector{} // init
+
+	for _, group := range recipe.Spec.Groups {
+		if group.Name == name {
+			labelSelector, err := getLabelSelectorFromString(group.LabelSelector)
```

Comment:
Originally the intention was to match the closed-source version of the Recipe with the RamenDR version field-for-field. We no longer have that requirement, but this is an artifact of that original decision. No specific code reasons exist to keep this as-is.

Reply:
It would be better to support these as label selectors (in other places in the Recipe as well). Tracked here https://github.com/RamenDR/recipe/issues/5